File: java/src/test/java/problems101_200/MaximumDepthOfBinaryTreeTest.java
package problems101_200;

import concepts.TreeNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class MaximumDepthOfBinaryTreeTest extends SampledTest {
    @Test
    public void test() {
        for (MaximumDepthOfBinaryTree solution : MaximumDepthOfBinaryTree.values()) {
            for (MaximumDepthOfBinaryTreeSample sample
                    : getSamples(MaximumDepthOfBinaryTreeSample[].class)) {
                assertWithMessage(sample.getMessage())
                        .that(solution.maxDepth(TreeNode.of(sample.input.root)))
                        .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/MaximumDepthOfBinaryTreeTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import concepts.TreeNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class MaximumDepthOfBinaryTreeTest : SampledTest() {
    @Test
    fun test() =
        MaximumDepthOfBinaryTree.values().forEach { solution ->
            getSamples<Array<MaximumDepthOfBinaryTreeSample>>()
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.maxDepth(TreeNode.of(*input.root)))
                        .isEqualTo(output)
                }
        }
}
